### PR TITLE
Simplify user endpoints

### PR DESCRIPTION
**Pull request by Frecherenkel60**
## Description

- Simplify endpoints and omit UserID in favour of the extracted UserID from the JWT

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

- [x] Locally tested the according endpoints
- [ ] Ran E2E-Tests

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by 930C — APPROVED** [review]

**Review by 930C — APPROVED** [review]
lgtm

File: src/controller/userController.go [hunk 1]
```diff
@@ -75,37 +74,50 @@ func (uc *UserController) RegisterUser(ctx context.Context, registrationData mod
...
+		return expenseerror.EXPENSE_UPSTREAM_ERROR
 	}
 
-	response := &model.RegistrationResponse{
-		UserID: &userId,
+	activationMail := &model.ActivationMail{
+		Username:        user.UserName,
+		ActivationToken: activationToken,
+		Subject:         activationMailSubject,
+		Recipients:      []string{user.Email},
```
Comment (930C) [inline src/controller/userController.go:112]: Array?

Comment (Frecherenkel60) [reply src/controller/userController.go:112]: Die API von Mailgun möchte einen Slice haben, also übergeb ich es hier auch so

**Review by 930C — APPROVED** [review]
l
g
t
m

**Pull request merged**